### PR TITLE
[wxfile] impl inline array threshold

### LINKED ISSUE #601: set default inline threshold
I think it would make sense to set a (sensible) default threshold to inline array values for most of our use cases

I think 10 should be a good value because it will inline a single (static) 3x rotation matrix and short list of vectors e.g. movements between two points
```yaml
coordinates: !weldx!core/variable-0.1.0
    name: coordinates
    dimensions: [time, c]
    dtype: <f8
    data: !core/ndarray-1.0.0
      data:
      - [20.0, 0.0, 3.0]
      - [330.0, 0.0, 3.0]
      datatype: float64
      shape: [2, 3]
```

### PR DESCRIPTION
**Pull request by marscher**
## Changes

ASDF 2.8 introduced the config context and deprecated this as a plain argument. However in the wrapper I think it'd be a sane solution to have it as an (optional) parameter again. It is delegated to the temporary config context.

## Related Issues

Closes #601 

## Checks

- [x] updated CHANGELOG.rst
- [x] updated tests
- [ ] updated doc/
- [ ] update example/tutorial notebooks
- [ ] update manifest file

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,12 @@
 added
 =====
 
+- `WeldxFile` handles an ``array_inline_threshold`` parameter to indicate if short arrays
+  will be serialized as strings, or as binary block. Note that this not affect arrays,
```
Comment (CagtayFabry) [inline CHANGELOG.rst:13]:
```suggestion
  will be serialized as strings, or as binary block. Note that this does not affect arrays,
```

File: weldx/asdf/file.py [hunk 2]
```diff
@@ -300,6 +312,21 @@ def __init__(
...
+            "array_inline_threshold" not in kwargs
+            or kwargs["array_inline_threshold"] is None
+        ):
+            kwargs["array_inline_threshold"] = self._array_inline_threshold
+
+        with config_context() as config:
+            for k, v in kwargs.items():
+                setattr(config, k, v)
+            yield
+
```
Comment (CagtayFabry) [inline weldx/asdf/file.py:329]: If I understand this correctly all (internal) `WeldxFile` operations should now use this custom context? I think this is a very good solution and more in line with the new asdf API 👍

**Review by CagtayFabry — APPROVED** [review]
LGTM 🚀 
should be a nice addition to the `0.5.2` release

File: weldx/asdf/file.py [hunk 3]
```diff
@@ -139,6 +143,10 @@ class WeldxFile(_ProtectedViewDict):
         When `False`, when reading files, attempt to memory map (memmap) underlying data
         arrays when possible. This avoids blowing the memory when working with very
         large datasets.
+    array_inline_threshold :
+        arrays below this threshold will be serialized as string, if larger as binary
+        block. Note that this not affect arrays, which are being shared across several
+        objects in the same file.
```
Comment (vhirtham) [inline weldx/asdf/file.py:149]:
```suggestion
        block. Note that this does not affect arrays, which are being shared across several
        objects in the same file.
```
Same as above (already mentioned by @CagtayFabry )

**Review by vhirtham — APPROVED** [review]

File: weldx/asdf/file.py (hunk 2 above)
Comment (marscher) [reply weldx/asdf/file.py:329]: Yes, currently it is only used where really needed. That includes write operations. But I guess later on, we will wrap everything in this context.

**Pull request merged**